chromium/pages/popup/onion-rules.js

```diff
@@ -0,0 +1,38 @@
+function e(id) {
+return document.getElementById(id);
+}
+  
+  /**
+ * Fill in content into the popup on load
+ */
+document.addEventListener("DOMContentLoaded", function () {
+  // auto-translate all elements with i18n attributes
+  var elem = document.querySelectorAll("[i18n]");
+  for (var i=0; i < elem.length; i++) {
+    elem[i].innerHTML = chrome.i18n.getMessage(elem[i].getAttribute("i18n"));
+  }
+
+  showRules();
+});
+
+function _createTD(text) {
+  var td = document.createElement('td');
+  var textNode = document.createTextNode(text);
+  td.appendChild(textNode);
+  return td;
+}
+
+function showRules() {
+  chrome.runtime.sendMessage({type: "get_simple_rules_ending_with", object: ".tor.onion"}, rules => {
+    var tbody = document.getElementById('onionTable');
+    for (var i in rules) {
+      var rule = rules[i];
+      var tr = document.createElement("tr");
+      tr.appendChild(_createTD(rule.from_regex));
+      tr.appendChild(_createTD(rule.scope_regex));
+      tr.appendChild(_createTD(rule.host));
+      tr.appendChild(_createTD(rule.to));
+      tbody.appendChild(tr);
+    }
+  });
+}
```

Comment:
Please add a newline at the end of this file.

Reply:
Done.